cpp/src/arrow/dataset/partition.cc

```diff
@@ -61,6 +61,16 @@ Result<std::string> SafeUriUnescape(util::string_view encoded) {
   }
   return decoded;
 }
+
+std::string StripNonPrefix(const std::string& path) {
+  std::string v;
+  auto non_prefix_index = path.rfind(kFilenamePartitionSep);
+  if (path.length() > 0 && non_prefix_index != std::string::npos) {
```

Comment:
```suggestion
  if (non_prefix_index != std::string::npos) {
```
Minor nit: If `length() <= 0` then `non_prefix_index != std::string::npos` will always be false so you can do away with the first check.

Reply:
Made the change.